keg_bouncer/model/migration.py

```diff
@@ -21,3 +24,68 @@ def link_user_to_user_groups(op, user_primary_key_column):
 
 def drop_link_from_user_to_user_groups(op):
     op.drop_table(user_user_group_link_table_name)
+
+
+def alembic_create_permission(op, permissions=[{}], groups=set(), bundles=set()):
+    """Bulk create permissions and add them to groups and bundles
+
+    :param permisisons: a list of dictionaries with 'token' and 'description' keys
+    :param groups: a set of group labels, missing groups will be created
+    :param bundles: a set of bundle labels, missing bundles will be created
+    """
+
+    new_permission_tokens = set([perm['token'] for perm in permissions])
+    groups = set(groups)
+    bundles = set(bundles)
+    con = op.get_bind()
+    meta = sa.MetaData()
+    execute = con.execute
+
+    # inspect the tables we need and create a table object for them
+    permissions_table = sa.Table('keg_bouncer_permissions', meta, autoload=True,
+                                 autoload_with=con.engine)
+
+    groups_table = sa.Table('keg_bouncer_user_groups', meta, autoload=True,
+                            autoload_with=con.engine)
+
+    groups_map = sa.Table('keg_bouncer_user_group_permission_map', meta, autoload=True,
+                          autoload_with=con.engine)
+
+    bundles_table = sa.Table('keg_bouncer_permission_bundles', meta, autoload=True,
+                             autoload_with=con.engine)
+
+    bundles_map = sa.Table('keg_bouncer_bundle_permission_map', meta, autoload=True,
+                           autoload_with=con.engine)
```

Comment:
> However, that only helps if you change the name/type of the column somewhere in the migration after this code runs. 

Including all future migrations. Remember, your DB can be at one state and mine can be at another. 

> If your tables have already changed, then you can't use this migration because it hard-codes the column names.
Correct, this function (and really any function for alembic that assumed names) will only work on tables that actually have those names, but that is the table, not the ORM. 

> In that case, using the ORM would have been better.
It is still just as bad. Remember, alembic runs in the checked out commit so it will use the code from the current source code. Inspecting the table allows you to separate from the code base and use whatever is in the database right now.